inlong-audit/audit-proxy/src/main/java/org/apache/inlong/audit/sink/KafkaSink.java

```diff
@@ -0,0 +1,430 @@
+
+    private boolean overflow = false;
+
+    private String localIp = "127.0.0.1";
+
+    static {
+        // stat kafka performance
+        logger.info("init kafkaPerformanceTask");
+        scheduledExecutorService.scheduleWithFixedDelay(kafkaPerformanceTask, 0L,
+                PRINT_INTERVAL, TimeUnit.SECONDS);
+    }
+
+    public KafkaSink() {
+        super();
+        logger.debug("new instance of KafkaSink!");
+    }
+
+    @Override
+    public synchronized void start() {
+        logger.info("kafka sink starting");
+        // create connection
+
+        sinkCounter.start();
+        super.start();
+        this.canSend = true;
+        this.canTake = true;
+
+        // init topic producer
+        initTopicProducer(topic);
+
+        for (int i = 0; i < sinkThreadPool.length; i++) {
+            sinkThreadPool[i] = new Thread(new SinkTask(), getName() + "_tube_sink_sender-" + i);
+            sinkThreadPool[i].start();
+        }
+        logger.debug("kafka sink started");
+    }
+
+    @Override
+    public synchronized void stop() {
+        logger.info("kafka sink stopping");
+        // stop connection
+        this.canTake = false;
+        int waitCount = 0;
+        while (eventQueue.size() != 0 && waitCount++ < 10) {
+            try {
+                Thread.sleep(1000);
+            } catch (InterruptedException e) {
+                logger.info("Stop thread has been interrupt!");
+                break;
+            }
+        }
+        this.canSend = false;
+
+        if (sinkThreadPool != null) {
+            for (Thread thread : sinkThreadPool) {
+                if (thread != null) {
+                    thread.interrupt();
+                }
+            }
+            sinkThreadPool = null;
+        }
+        super.stop();
+        if (!scheduledExecutorService.isShutdown()) {
+            scheduledExecutorService.shutdown();
+        }
+        sinkCounter.stop();
+        logger.debug("kafka sink stopped. Metrics:{}", sinkCounter);
+    }
+
+    @Override
+    public Status process() {
+        logger.info("kafka sink processing");
+        if (!this.canTake) {
+            return Status.BACKOFF;
+        }
+
+        Status status = Status.READY;
+        Channel channel = getChannel();
+        Transaction tx = channel.getTransaction();
+        tx.begin();
+        try {
+            Event event = channel.take();
+            if (event != null) {
+                if (diskRateLimiter != null) {
+                    diskRateLimiter.acquire(event.getBody().length);
+                }
+                if (!eventQueue.offer(event, 3 * 1000, TimeUnit.MILLISECONDS)) {
+                    logger.info("[{}] Channel --> Queue(has no enough space,current code point) "
+                            + "--> Kafka,Check if Kafka server or network is ok.(if this situation "
+                            + "last long time it will cause memoryChannel full and fileChannel write.)", getName());
```

Comment:
```suggestion
                    logger.info("[{}] Channel --> Queue(not enough space, current code point) "
                            + "--> Kafka, check if Kafka server or network is ok. (If this situation "
                            + "last long time it will cause memoryChannel full and fileChannel write)", getName());
```

Reply:
Done.